### PR TITLE
[Backend] Add feature: manage subscriptions

### PR DESCRIPTION
**Pull request by rasouza**
## Description
- Implement `/subscriptions` RESTful resources
- Improve unit testing by implementing automock
- Add global error handler
- Add Zod serializer
- Update package.json to ensure `volta` can fetch the correct tools from monorepo

#### Unit testing with automock
We no longer need to recreate the app module during unit tests anymore (see [here](https://github.com/mathiasberggren/streaming-subscription/pull/11/files#diff-0e9c23efd0687eab382d17b8a2c0a6716ab4aaa9a2b82f479d7ecbe808dec217R26-R35)). Easier to mock and write tests. You can still retrieve any dependency with `unit` and `unitRef` ([official doc](https://docs.nestjs.com/recipes/automock))

#### Zod serializer
You can think of this as the _Presenter_ layer of the Clean Architecture. In short, neither controller nor services should be in charge of presenting the response payload to clients. Instead, we introduce tools to make it easier to present the response the way we need, hence the Serializers. 

NestJS has a builtin for that as well ([doc](https://docs.nestjs.com/techniques/serialization)), but since we are using Zod and it has support for serializing, we can stick to that lib

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: apps/api/src/database/schema.prisma [hunk 1]
```diff
@@ -14,7 +14,53 @@ datasource db {
...
+  subscriptions Subscription[]
+
+  @@map("users")
+}
+
+model Company {
+  id        Int      @id @default(autoincrement())
+  name      String
+  image     String
+  createdAt DateTime @default(now()) @map("created_at")
```
Comment (mathiasberggren) [inline apps/api/src/database/schema.prisma:32]: Is there a reason why we name these with camelCase and map them to a snakeCase? Similar to how we do with DTO's in Klarna to validate when data is confirmed to be type safe?

File: apps/api/src/app/app.module.ts [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
   controllers: [AppController],
   providers: [
     Logger,
     AppService,
     {
       provide: APP_PIPE,
       useClass: ZodValidationPipe
+    },
+    {
+      provide: APP_INTERCEPTOR,
```
Comment (mathiasberggren) [inline apps/api/src/app/app.module.ts:31]: So what we're doing here is basically addding a validation middleware of the incoming payload? Or?

File: apps/api/src/app/http-exception.filter.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { ExceptionFilter, Catch, ArgumentsHost, HttpException, Logger } from '@nestjs/common'
+import { Request, Response } from 'express'
+
+@Catch()
+export class HttpExceptionFilter implements ExceptionFilter {
```
Comment (mathiasberggren) [inline apps/api/src/app/http-exception.filter.ts:5]: So if I understood correctly this is an error handling middleware?

**Review by mathiasberggren — APPROVED** [review]

File: apps/api/src/database/schema.prisma (hunk 1 above)
Comment (rasouza) [reply apps/api/src/database/schema.prisma:32]: Oof nice catch! This is because databases usually has a different naming convention (plural and snake_case) from JS naming convention (singular and camelCase) and we want both to coexist.

This is described in more details in this [Prisma doc](https://www.prisma.io/docs/orm/prisma-schema/data-model/database-mapping). Schema file gets a little bit more verbose but code and database is standardized in a nice way

In short, in the left you read as field names while coding (client generated by Prisma) and on the right (the @map field) is how you will see in the database tables

File: apps/api/src/app/app.module.ts (hunk 2 above)
Comment (rasouza) [reply apps/api/src/app/app.module.ts:31]: incoming payload validation would be the block above (`APP_PIPE` and `ZodValidationPipe`), this one is about how you want response payload to be shown

File: apps/api/src/app/http-exception.filter.ts (hunk 3 above)
Comment (rasouza) [reply apps/api/src/app/http-exception.filter.ts:5]: Precisely! The framework has a nice built-in global error handler already, but it was omitting errors from Zod schemas. I had no idea what was failing so I wrote another one myself, just to `console.log` the error

In the end, it helped me to identify that I was typing `.optional()` instead of `.nullable()`
